### PR TITLE
Move get_transaction_from_request to client to allow overrides

### PR DESCRIPTION
**Pull request by jsferrei**
@dcramer @mitsuhiko this is a redo of https://github.com/getsentry/raven-python/pull/911
-- I went to fix conflicts and severely messed up history, so figured it better to recreate as new

### CODE REVIEW

**Comment by mitsuhiko** [comment]
From what I can tell this looks okay. Do we want this @dcramer?

File: raven/contrib/django/client.py [hunk 1]
```diff
@@ -297,3 +299,7 @@ def capture(self, event_type, request=None, **kwargs):
             }
 
         return result
+
+    def get_transaction_from_request(self, request):
+        result = self.resolver.resolve(request.path)
```
Comment (mattrobenolt) [inline raven/contrib/django/client.py]: Can we skip the unnecessary variable here please?

**Comment by jsferrei** [comment]
@dcramer @mitsuhiko Any update on this/requested changes? Would like this so we can override the default and leverage sentry on our app stack.

**Comment by mattrobenolt** [comment]
Thanks! 🥇

**Comment by jsferrei** [comment]
Thanks!

**Pull request merged**